tests/admin_utils/test_logentry.py

```diff
@@ -248,7 +336,7 @@ def test_recentactions_without_content_type(self):
         link = reverse("admin:admin_utils_article_change", args=(quote(self.a1.pk),))
         should_contain = """<a href="%s">%s</a>""" % (
             escape(link),
-            escape(repr(self.a1)),
+            escape(str(self.a1)),
```

Comment:
This looks unrelated (but it's required), and makes me feel that we're still making some backward incompatible change here.

Reply:
There is one backward incompatible change here.
Look at the earlier `log_deletion` function,
```
def log_deletion(self, request, obj, object_repr):
        """
        Log that an object will be deleted. Note that this method must be
        called before the deletion.

        The default implementation creates an admin LogEntry object.
        """
        from django.contrib.admin.models import DELETION, LogEntry

        return LogEntry.objects.log_action(
            user_id=request.user.pk,
            content_type_id=get_content_type_for_model(obj).pk,
            object_id=obj.pk,
            object_repr=object_repr,
            action_flag=DELETION,
        )
```

The `object_repr` is being passed as a parameter here for each and every object.

Now look at the new `log_deletions` method,
```
    def log_deletions(self, request, queryset, message=""):
        """
        Log that multiple objects will be deleted. Note that this method must
        be called before the deletion of multiple objects.

        The default implementation creates some LogEntry objects.
        """

        # If log_deletion is overridden in a subclass, call it to maintain
        # backward compatibility
        if type(self).log_deletion != ModelAdmin.log_deletion:
            return [self.log_deletion(request, obj, str(obj)) for obj in queryset]

        from django.contrib.admin.models import DELETION, LogEntry

        return LogEntry.objects.log_actions(
            user_id=request.user.pk,
            queryset=queryset,
            action_flag=DELETION,
            change_message=message,
        )
```

Here we are not passing the `object_repr`  parameter. It is generated automatically within `log_actions`

Reply:
`log_actions()` uses `str()`, and `log_deletion()` before this change also used `str()`, 

https://github.com/django/django/blob/3b255d72cf73e89ac1dc94d74467e5e4c487f5f2/django/contrib/admin/options.py#L2174-L2177

https://github.com/django/django/blob/3b255d72cf73e89ac1dc94d74467e5e4c487f5f2/django/contrib/admin/actions.py#L48-L50

So I still don't understand why this was set to `repr()` and why needs to be changed now.

Reply:
Ah, OK now I understand. It's caused by `log_actions()` in `setUpTestData()`.